### PR TITLE
Add after test hook

### PR DESCRIPTION
**Pull request by lvijnck**
Solves #759 

I noticed the other PR or a more complicated mechanism to add before/after hooks. Though I really need this functionality as I'm running a large number of tests and I'm unable to perform proper cleanup now.

Happy to make changes as required.

### CODE REVIEW

File: tavern/testutils/pytesthook/item.py [hunk 1]
```diff
@@ -194,6 +194,16 @@ def runtest(self):
                     stage["name"] = stage["id"]
 
             run_test(self.path, self.spec, self.global_cfg)
+
+            # TODO: Will this also run after test failure?
```
Comment (michaelboulton) [inline tavern/testutils/pytesthook/item.py:198]: This won't be run if the test fails, you would need to put it in a 'finally' block or that to work.

**Review by michaelboulton — CHANGES_REQUESTED** [review]
This is fine, but could you also just add a bit a documentation at the bottom of `docs/source/basics.md` near the other documentation for the hooks?

**Comment by lvijnck** [comment]
> This is fine, but could you also just add a bit a documentation at the bottom of `docs/source/basics.md` near the other documentation for the hooks?

Done! LMK if any other changes are required.

P.S. I was checking whether there are any unit tests added for these hooks, but I could not directly find them.

**Comment by michaelboulton** [comment]
> > This is fine, but could you also just add a bit a documentation at the bottom of `docs/source/basics.md` near the other documentation for the hooks?
> 
> Done! LMK if any other changes are required.
> 
> P.S. I was checking whether there are any unit tests added for these hooks, but I could not directly find them.

There aren't any tests for it as they only get run when you run a 'real test due to the way pytest works, if the integration tests pass then you can assume that the hook is being called correctly

**Review by michaelboulton — APPROVED** [review]

**Pull request merged**